pylint/checkers/typecheck.py

```diff
@@ -1290,11 +1312,16 @@ def _is_ignored_function(
         )
 
     @staticmethod
-    def _is_list_sort_method(node: nodes.Call) -> bool:
+    def _is_builtin_no_return(node: nodes.Assign) -> bool:
         return (
-            isinstance(node.func, nodes.Attribute)
-            and node.func.attrname == "sort"
-            and isinstance(utils.safe_infer(node.func.expr), nodes.List)
+            isinstance(node.value, nodes.Call)
+            and isinstance(node.value.func, nodes.Attribute)
+            and bool(inferred := utils.safe_infer(node.value.func.expr))
+            and isinstance(inferred, bases.Instance)
+            and bool(pytype := inferred.pytype())
+            and pytype.startswith("builtins")
```

Comment:
This is a bit redundant with line 1323 right?
We could even do a `.get(pytype, ())` but I don't know if you like that style.

Reply:
Oh good catch that I can remove the startswith. Remnant of earlier version of PR.

> We could even do a .get(pytype, ()) but I don't know if you like that style.

Sure, I can do that.